### ISSUE TITLE
Relay does not forward envelopes to sentry

### ISSUE THREAD

**Issue by sn0wk4t** [getsentry/relay#3034]
Hello,

I'm running a custom sentry deployment on VMs (currently at aversion `23.6.2`) based on the self-hosted solution.
Everything seems to work fine except for the relay container where I see bunch of these logs

```
2024-01-31T15:48:24.655122Z ERROR relay_server::actors::global_config: global config missing in upstream response
2024-01-31T15:48:34.696067Z ERROR relay_server::actors::global_config: global config missing in upstream response
2024-01-31T15:48:44.736353Z ERROR relay_server::actors::global_config: global config missing in upstream response
2024-01-31T15:48:54.770836Z ERROR relay_server::actors::global_config: global config missing in upstream response
2024-01-31T15:49:04.806852Z ERROR relay_server::actors::global_config: global config missing in upstream response
2024-01-31T15:49:14.840232Z ERROR relay_server::actors::global_config: global config missing in upstream response
2024-01-31T15:49:24.883324Z ERROR relay_server::actors::global_config: global config missing in upstream response
2024-01-31T15:49:34.921620Z ERROR relay_server::actors::global_config: global config missing in upstream response
2024-01-31T15:49:44.971518Z ERROR relay_server::actors::global_config: global config missing in upstream response
2024-01-31T15:49:55.010902Z ERROR relay_server::actors::global_config: global config missing in upstream response
2024-01-31T15:50:00.239808Z ERROR relay_server::endpoints::common: error handling request error=failed to queue envelope error.sources=[envelope buffer capacity exceeded]
2024-01-31T15:50:00.241839Z ERROR relay_server::endpoints::common: error handling request error=failed to queue envelope error.sources=[envelope buffer capacity exceeded]
2024-01-31T15:50:00.243445Z ERROR relay_server::endpoints::common: error handling request error=failed to queue envelope error.sources=[envelope buffer capacity exceeded]
2024-01-31T15:50:00.243877Z ERROR relay_server::endpoints::common: error handling request error=failed to queue envelope error.sources=[envelope buffer capacity exceeded]
2024-01-31T15:50:00.245600Z ERROR relay_server::endpoints::common: error handling request error=failed to queue envelope error.sources=[envelope buffer capacity exceeded]
```

For what I've seen it looks like relay tries to fetch project configs via `POST /api/0/relays/projectconfigs/?version=3`
but an empty payload is returned
```
{"configs":{}}
```
then envelopes begin to spool in relay memory eventually reaching it's limit. Thus I get this error and the event is dropped
```
2024-01-31T15:17:09.197539Z ERROR relay_server::utils::managed_envelope: dropped envelope: internal error tags.has_attachments=false tags.has_sessions=false tags.has_profiles=false tags.has_transactions=false tags.has_replays=false tags.has_checkins=false tags.event_category=Some(Transaction) cached_summary=EnvelopeSummary { event_category: Some(Transaction), attachment_quantity: 0, session_quantity: 0, profile_quantity: 0, replay_quantity: 0, checkin_quantity: 0, secondary_transaction_quantity: 0, has_plain_attachments: false, event_metrics_extracted: false, payload_size: 5851 } recomputed_summary=EnvelopeSummary { event_category: Some(Transaction), attachment_quantity: 0, session_quantity: 0, profile_quantity: 0, replay_quantity: 0, checkin_quantity: 0, secondary_transaction_quantity: 0, has_plain_attachments: false, event_metrics_extracted: false, payload_size: 5851 }
```

I already checked the sentry and snuba workers and didn't find any error except for the sentry worker where I get lots of
`[ERROR] sentry_sdk.errors: Unexpected status code: 503 (body: b'{"detail":"failed to queue envelope","causes":["envelope buffer capacity exceeded"]}')`. 
Looks like other workers are simply sitting still in waiting.
No activity on the celery side either
```
# docker exec -it sentry_web sentry queues list
15:35:04 [WARNING] sentry.utils.geo: Error opening GeoIP database: /geoip/GeoLite2-City.mmdb
15:35:04 [WARNING] sentry.utils.geo: Error opening GeoIP database in Rust: /geoip/GeoLite2-City.mmdb
activity.notify 0
alerts 0
app_platform 0
app_platform.control 0
appstoreconnect 0
assemble 0
auth 0
auto_enable_codecov 0
auto_transition_issue_states 0
buffers.incr 0
buffers.process_pending 0
cleanup 0
code_owners 0
commits 0
counters-0 0
data_export 0
default 0
derive_code_mappings 0
digests.delivery 0
digests.scheduling 0
dynamicsampling 0
email 0
events.preprocess_event 0
events.process_event 0
events.reprocess_events 0
events.reprocessing.preprocess_event 0
events.reprocessing.process_event 0
events.reprocessing.symbolicate_event 0
events.reprocessing.symbolicate_event_low_priority 0
events.save_event 0
events.save_event_attachments 0
events.save_event_transaction 0
events.symbolicate_event 0
events.symbolicate_event_low_priority 0
events.symbolicate_js_event 0
events.symbolicate_js_event_low_priority 0
files.copy 0
files.delete 0
get_suspect_resolutions 0
get_suspect_resolutions_releases 0
group_owners.process_commit_context 0
group_owners.process_suspect_commits 0
hybrid_cloud.control_repair 0
incident_snapshots 0
incidents 0
incidents 0
integrations 0
merge 0
options 0
post_process_errors 0
post_process_issue_platform 0
post_process_transactions 0
profiles.process 0
recap_servers 0
relay_config 0
relay_config_bulk 0
releasemonitor 0
replays.delete_replay 0
replays.ingest_replay 0
reports.deliver 0
reports.prepare 0
search 0
sentry_metrics.indexer 0
similarity.index 0
sleep 0
stats 0
subscriptions 0
symbolications.compute_low_priority_projects 0
transactions.name_clusterer 0
triggers-0 0
unmerge 0
update 0
weekly_escalating_forecast 0
```

I also don't see any issues with my kafka or redis cluster.
This is my relay configuration
```
---
relay:
  upstream: "http://10.15.14.90:80"
  host: 0.0.0.0
  port: 3000
  tls_port: ~
  tls_identity_path: ~
  tls_identity_password: ~
logging:
  level: WARN
processing:
  enabled: true
  kafka_config:
    - {name: "bootstrap.servers", value: "10.15.14.254:9092"}
    - {name: "message.max.bytes", value: 50000000} # 50MB
  redis: redis://10.15.14.243:6379
```
Any hints on how to debug this?

**Comment by jjbayer** [comment]
Hey @sn0wk4t, is your relay version newer than 23.6.2? Newer relay versions require a global config from sentry, but it is only served by newer sentry versions.

TL;DR align the version tags on relay and sentry.

**Comment by sn0wk4t** [comment]
Hi @jjbayer thanks for the response and for pin pointing the issue!
I was indeed pulling the latest tag of relay due to an ansible error while sentry, snuba and vroom images version were at 23.6.2.
With the proper relay version everything started to work again
